packages/formstr-app/src/containers/CreateForm/providers/FormBuilder/index.tsx

```diff
@@ -50,9 +66,32 @@ export default function FormBuilderProvider({
     setQuestionsList([...questionsList, generateQuestion(answerType)]);
   };
 
+  const deleteQuestion = (tempId: string) => {
+    if (questionIdInFocus === tempId) {
+      setQuestionIdInFocus(undefined);
+    }
+    setQuestionsList((preQuestions) => {
+      return preQuestions.filter((question) => question.tempId !== tempId);
+    });
+  };
+
+  const updateFormSetting = (settings: IFormSettings) => {
+    setFormSettings((preSettings) => ({ ...preSettings, ...settings }));
+  };
+
   return (
```

Comment:
just thinking out loud here. We do we need a provider-context in general.

Just storing the state and exposing the state/updater functions directly can work as well right?

For eg.

```ts
// useSettings
const [settings, updateSettings] = React.useState({})

return {
settings,
updateSettings
}

//Question
const settings = useSettings()
...
```

What advantage is Provider-Context giving us here?

Reply:
added setting functionality in context here just to make sure it doesn't remove existing property